src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2498,26 +2498,12 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetSrpDnsCallba
 template <class ImplClass>
 CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsResponseToMdnsData(
     otDnsServiceInfo & serviceInfo, const char * serviceType, chip::Dnssd::DnssdService & mdnsService,
-    DnsServiceTxtEntries & serviceTxtEntries)
+    DnsServiceTxtEntries & serviceTxtEntries, otError error)
 {
     char protocol[chip::Dnssd::kDnssdProtocolTextMaxSize + 1];
 
-    if (strchr(serviceInfo.mHostNameBuffer, '.') == nullptr)
-        return CHIP_ERROR_INVALID_ARGUMENT;
-
-    // Extract from the <hostname>.<domain-name>. the <hostname> part.
-    size_t substringSize = strchr(serviceInfo.mHostNameBuffer, '.') - serviceInfo.mHostNameBuffer;
-    if (substringSize >= ArraySize(mdnsService.mHostName))
-    {
-        return CHIP_ERROR_INVALID_ARGUMENT;
-    }
-    Platform::CopyString(mdnsService.mHostName, serviceInfo.mHostNameBuffer);
-
-    if (strchr(serviceType, '.') == nullptr)
-        return CHIP_ERROR_INVALID_ARGUMENT;
-
     // Extract from the <type>.<protocol>.<domain-name>. the <type> part.
-    substringSize = strchr(serviceType, '.') - serviceType;
+    size_t substringSize = strchr(serviceType, '.') - serviceType;
```

Comment:
This lost the check that makes this safe:
```
    if (strchr(serviceType, '.') == nullptr)
        return CHIP_ERROR_INVALID_ARGUMENT;
```
because that looks like it was incorrectly moved to the block below.